### PR TITLE
Fix integer constant overflows in tests

### PR DESCRIPTION
**Pull request by menschel-d**
Explicitly use int64 in order to avoid the following errors on 32 bit
architectures:

```
# code.gitea.io/gitea/integrations
./api_admin_test.go:50:34: constant 9223372036854775807 overflows int
./api_token_test.go:47:34: constant 9223372036854775807 overflows int
[...]
# code.gitea.io/gitea/models
./action_test.go:179:15: constant 9223372036854775807 overflows int
```

Futhermore, the magic constant 9223372036854775807 has been substituted with a numerically equivalent integer limit value.

---

I'm currently working on a BitBake recipe for Gitea in the context of Yocto/OpenEmbedded, and have stumbled upon the mentioned problems when compiling for the `qemuarm` target.
As I have little experience in Go programming, I've adapted the solution from [1].
If there are any problems in the code changes, just let me know.

[1] https://stackoverflow.com/a/16475045

### CODE REVIEW

File: models/unit_tests.go [hunk 1]
```diff
@@ -23,7 +24,7 @@ import (
 )
 
 // NonexistentID an ID that will never exist
-const NonexistentID = 9223372036854775807
+const NonexistentID = math.MaxInt64
```
Comment (sapk) [inline models/unit_tests.go]: Can you simply do `const NonexistentID = int64(math.MaxInt64)` and don't edit others code part ?

**Review by sapk — CHANGES_REQUESTED** [review]

**Comment by menschel-d** [comment]
Thanks for the suggestion,  @sapk.
These changes should now also be easier to backport to the latest stable release.

File: .gitignore [hunk 2]
```diff
@@ -69,3 +69,7 @@ prime/
 *.snap
 *.snap-build
 *_source.tar.bz2
+
+
+# Backup files (i.e. from Vim)
```
Comment (techknowlogick) [inline .gitignore]: This should belong in your global git ignore file, or at least create a new PR for it as it is unrelated to the current PR.

**Comment by menschel-d** [comment]
Thanks for the quick reply, @techknowlogick. I've omitted the changes to `.gitignore` like you've requested.

**Comment by techknowlogick** [comment]
Thanks @menschel-d 

LGTM

**Review by sapk — APPROVED** [review]
Thanks for the contrib.

**Review by lafriks — APPROVED** [review]

**Pull request merged**